testcert.py

```diff
@@ -1957,310 +2170,297 @@ def testPrinterOffSendGoodbyePacket(self):
     test_name = 'testPrinterOffSendGoodbyePacket'
     failed = False
     printer_found = False
+
     print 'This test must start with the printer on and operational.'
-    raw_input('Power off printer, Select enter when printer completely off.')
-    time.sleep(10)
-    for v in mdns_browser.listener.discovered.values():
+    PromptUserAction('Turn off the printer and wait...')
+    is_removed = waitForService(_device.name, False, timeout=300)
+    try:
+      self.assertTrue(is_removed)
+    except AssertionError:
+      notes = 'Error receiving the shutdown signal from the printer.'
+      self.LogTest(test_id, test_name, 'Failed', notes)
+      raise
+
+    for v in _mdns_browser.listener.discovered.values():
       if 'ty' in v['info'].properties:
         if self.printer in v['info'].properties['ty']:
           printer_found = True
           try:
             self.assertFalse(v['found'])
+            break
           except AssertionError:
-            notes = 'Printer did not send goodbye packet when powered off.'
             failed = True
-            raise
-          else:
-            notes = 'Printer sent goodbye packet when powered off.'
     if not printer_found:
-      notes = 'Printer did not send goodbye packet when powered off.'
       failed = True
 
     if failed:
+      notes = 'Printer did not send goodbye packet when powered off.'
       self.LogTest(test_id, test_name, 'Failed', notes)
+      raise
     else:
+      notes = 'Printer sent goodbye packet when powered off.'
       self.LogTest(test_id, test_name, 'Passed', notes)
 
+    PromptUserAction('Power on the printer and wait...')
```

Comment:
It might be nice for the user to know how long the script is about to (potentially) wait for.